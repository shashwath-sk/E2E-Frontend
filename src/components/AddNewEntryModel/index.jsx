import React from 'react';
import './EntryModal.css';
import PropTypes from 'prop-types';
import axios from 'axios';
export default function AddNewEntryModel(props) {
    if (!props.show) {
        return null;
    }
    const handleAddEntries = async() => {
        await axios({
            method:'POST',
            url:`http://localhost:5000/api/addEntries/${props.contentId}`,
            data:{newfield:props.value}
        });
    };
    return (
        <div className="modal" onClick={props.onClose}>
            <div className="modal-content" onClick={e => e.stopPropagation()}>
                <div className='modal-header'>
                    <span className='modal-title'>New Company Profile</span>
                </div>
                <div className='modal-body'>
                    {
                        Object.keys(props.value).map((key,id) => {
                            return (
                                <div key={id}>
                                    <span>{key}</span>
                                    <input type="text" />
                                </div>
                            );
                        })
                    }
                </div>
                <div className='modal-footer'>
                    <button onClick={props.onClose} className='modal-close-button'>Close</button>
                    <button className='modal-Add-button' onClick={handleAddEntries}>Add</button>
                </div>
            </div>
        </div>
    );
}
AddNewEntryModel.propTypes = {
    onClose: PropTypes.func.isRequired,
    show: PropTypes.bool.isRequired,
    value: PropTypes.object,
    contentId: PropTypes.string
};