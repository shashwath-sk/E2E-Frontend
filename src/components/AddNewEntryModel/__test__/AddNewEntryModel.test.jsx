import AddNewEntryModel from '../';
import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';

describe('AddNewEntryModel', () => {
    it('should render AddNewEntryModel', () => {
        const { asFragment } = render(
            <BrowserRouter>
                <AddNewEntryModel />
            </BrowserRouter>
        );
        expect(asFragment).toMatchSnapshot();
    });
});