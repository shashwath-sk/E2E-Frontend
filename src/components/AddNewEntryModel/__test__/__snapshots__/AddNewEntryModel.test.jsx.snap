// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddNewEntryModel should render AddNewEntryModel 1`] = `[Function]`;
